File: solutions/3484-lexicographically-smallest-string-after-a-swap/lexicographically-smallest-string-after-a-swap.py
class Solution:
    def getSmallestString(self, s: str) -> str:
        arr = [int(ch) for ch in s]
        n = len(arr)

        for i in range(n):
            for j in range(1, n):
                if arr[j-1] % 2 == arr[j] % 2:
                    if arr[j-1] > arr[j]:
                        arr[j-1] = arr[j] + arr[j-1]
                        arr[j] = arr[j-1] - arr[j]
                        arr[j-1] = arr[j-1] - arr[j]
        
                        return "".join([str(num) for num in arr])
        return "".join([str(num) for num in arr])

File: solutions/3484-lexicographically-smallest-string-after-a-swap/lexicographically-smallest-string-after-a-swap.java
class Solution {
    public String getSmallestString(String s) {

        for(int i=0; i<s.length()-1; i++){
            int a = Character.getNumericValue(s.charAt(i));
            int b = Character.getNumericValue(s.charAt(i+1));

            if((a%2 == 0 && b%2 == 0 && b < a) || (a%2 != 0 && b%2!=0 && b < a)){
                StringBuilder sb = new StringBuilder(s);
                String ab = String.valueOf(a);
                String bb = String.valueOf(b);
                sb.setCharAt(i, bb.charAt(0));
                sb.setCharAt(i+1, ab.charAt(0));
                return sb.toString();
            }
        }
        return s;
    }
}